src/widgets/notes/notes-overview.component.tsx

```diff
@@ -1,106 +1,164 @@
+
+import {
+  Button,
+  DataTable,
+  DataTableSkeleton,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableHeader,
+  TableRow
+} from "carbon-components-react";
+import { Add16 } from "@carbon/icons-react";
+
 import { createErrorHandler } from "@openmrs/esm-error-handling";
-import { getEncounterObservableRESTAPI } from "./encounter.resource";
-import { formatNotesDate } from "./notes-helper";
-import SummaryCard from "../../ui-components/cards/summary-card.component";
-import SummaryCardFooter from "../../ui-components/cards/summary-card-footer.component";
-import EmptyState from "../../ui-components/empty-state/empty-state.component";
-import useChartBasePath from "../../utils/use-chart-base";
-import { PatientNotes } from "../types";
+import { useCurrentPatient } from "@openmrs/esm-react-utils";
+
 import { openWorkspaceTab } from "../shared-utils";
+import EmptyState from "../../ui-components/empty-state/empty-state.component";
+import ErrorState from "../../ui-components/error-state/error-state.component";
+
+import {
+  getEncounterObservableRESTAPI,
+  PatientNote
+} from "./encounter.resource";
+import { formatNotesDate } from "./notes-helper";
 import VisitNotes from "./visit-note.component";
-import styles from "./notes-overview.css";
+import styles from "./notes-overview.scss";
 
-export default function NotesOverview({ basePath }: NotesOverviewProps) {
-  const [patientNotes, setPatientNotes] = React.useState<Array<PatientNotes>>();
-  const [, patient, patientUuid] = useCurrentPatient();
-  const chartBasePath = useChartBasePath();
-  const notesPath = chartBasePath + "/" + basePath;
+const NotesOverview: React.FC<NotesOverviewProps> = () => {
   const { t } = useTranslation();
+  const [, patient, patientUuid] = useCurrentPatient();
+  const [notes, setNotes] = React.useState<Array<PatientNote>>(null);
+  const [error, setError] = React.useState(null);
+  const displayText = t("notes", "notes");
+  const headerTitle = t("notes", "Notes");
 
   React.useEffect(() => {
     if (patient && patientUuid) {
       const sub = getEncounterObservableRESTAPI(patientUuid).subscribe(
-        patientVisitNote => setPatientNotes(patientVisitNote),
-        createErrorHandler()
+        notes => setNotes(notes),
+        error => {
+          setError(error);
+          createErrorHandler();
+        }
       );
       return () => sub.unsubscribe();
     }
   }, [patient, patientUuid]);
 
+  const launchVisitNoteForm = () => {
+    openWorkspaceTab(VisitNotes, t("visitNotesForm", "Visit note form"));
+  };
+
+  const headers = [
+    {
+      key: "encounterDate",
+      header: t("date", "Date")
+    },
+    {
+      key: "encounterType",
+      header: t("encounterType", "Encounter type")
+    },
+    {
+      key: "encounterLocation",
+      header: t("location", "Location")
+    },
+    {
+      key: "encounterAuthor",
+      header: t("author", "Author")
+    }
+  ];
+
+  const getRowItems = rows =>

```

Comment:
Any reason for capturing this function or could we define it outside, too? What (type) is `rows`?

Reply:
In this case, the same as the `notes`, so type `PatientNote`. I defined it locally so I can make manipulations to the data coming in from the `note` object before it gets rendered. 

```
  const getRowItems = (rows: Array<PatientNote>) =>
    rows.map(row => ({
      ...row,
      encounterDate: formatNotesDate(row.encounterDate),
      author: row.encounterAuthor ? row.encounterAuthor : "\u2014"
    }));
```

Reply:
Yeah what I'm after is that we could have defined that in a function outside (thus making the render more lightweight).

```ts
function getRowItems(rows: Array<PatientNote>) {
  return rows.map(row => ({ ...row, ... }));
}
```

Reason is that `formatNotesDate` was available outside, so I did not see any captured value.

Reply:
Ah! Very clear. Thanks